### PR TITLE
[MTDSA-5116] RETRIEVE Controller, Routes & ITs

### PR DESCRIPTION
**Pull request by Mbrookesy**

### CODE REVIEW

File: app/v1/models/response/retrieve/RetrieveResponse.scala [hunk 1]
```diff
@@ -19,13 +19,14 @@ package v1.models.response.retrieve
 import play.api.libs.functional.syntax._
 import play.api.libs.json.{JsPath, Json, OWrites, Reads}
 
-case class RetrieveResponse(fromDate: String, toDate: String, periodData: Seq[PeriodDetails])
+case class RetrieveResponse(dateFrom: String, dateTo: String, periodData: Seq[PeriodDetails])
 
 object RetrieveResponse {
```
Comment (james-work-account) [inline app/v1/models/response/retrieve/RetrieveResponse.scala]: I'd prefer if this was the other way around, ie have the model how we represent the data and read it in with custom reads. Something like

```lang-scala
case class RetrieveResponse(fromDate: String, toDate: String, periodData: Seq[PeriodDetails])

object RetrieveResponse {
  implicit val reads: Reads[RetrieveResponse] = (
    (JsPath \ "dateFrom").read[String] and
      (JsPath \ "dateTo").read[String] and
      (JsPath \ "periodData").read[Seq[PeriodDetails]]
  )(RetrieveResponse.apply _)

  implicit val writes: OWrites[RetrieveResponse] = Json.writes[RetrieveResponse]
}
```
(you'll have to forgive any typos; I wrote this directly into gh not in an ide)

File: it/v1/endpoints/RetrieveControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+            val response: WSResponse = await(request().get())
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+        val input = Seq(
+          ("AA1123A", "S4636A77V5KB8625U", Status.BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "beans", Status.BAD_REQUEST, DeductionIdFormatError),
+          ("", "", Status.NOT_FOUND, NotFoundError)
```
Comment (james-work-account) [inline it/v1/endpoints/RetrieveControllerISpec.scala]: Is this error in the spec?

File: resources/public/api/conf/1.0/application.raml [hunk 3]
```diff
@@ -35,7 +35,8 @@ uses:
       nino:
         description: National Insurance number, in the format AA999999A.
         type: string
-        example: TC663795B
+        example:
+        TC663795B
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/application.raml]: Please put this back, it breaks the RAML.

File: test/v1/controllers/RetrieveControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (DeductionIdFormatError, BAD_REQUEST),
+          (BadRequestError, BAD_REQUEST),
+          (NotFoundError, NOT_FOUND),
```
Comment (james-work-account) [inline test/v1/controllers/RetrieveControllerSpec.scala]: Again, should the parser be returning NOT_FOUND? Without calling an external service, we technically can't validate whether what they send in can be found or not, so I wouldn't have expected this to be valid in the spec.

File: test/v1/controllers/RetrieveControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+          (DownstreamError, INTERNAL_SERVER_ERROR),
```
Comment (james-work-account) [inline test/v1/controllers/RetrieveControllerSpec.scala]: remove trailing comma

File: test/v1/controllers/RetrieveControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+
+          }
+        }
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (DeductionIdFormatError, BAD_REQUEST),
+          (BadRequestError, BAD_REQUEST),
+          (NotFoundError, NOT_FOUND),
+          (DownstreamError, INTERNAL_SERVER_ERROR),
```
Comment (james-work-account) [inline test/v1/controllers/RetrieveControllerSpec.scala]: remove trailing comma

File: it/v1/endpoints/RetrieveControllerISpec.scala (hunk 2 above)
Comment (Mbrookesy) [reply it/v1/endpoints/RetrieveControllerISpec.scala]: In the Spec it shows there being a FORMAT_NINO error, a FORMAT_ID error and a MATCHING_RESOURCE_NOT_FOUND error

File: test/v1/controllers/RetrieveControllerSpec.scala (hunk 4 above)
Comment (Mbrookesy) [reply test/v1/controllers/RetrieveControllerSpec.scala]: As far as I see there is a NOT_FOUND error within the Spec

Comment (james-work-account) [reply test/v1/controllers/RetrieveControllerSpec.scala]: The NOT_FOUND error is exclusively a backend thing

**Pull request merged**